tests/x11/firefox/firefox_html5.pm

```diff
@@ -5,41 +5,51 @@
 # SPDX-License-Identifier: FSFAP
 
 # Package: MozillaFirefox
-# Summary: Case#1479221: Firefox: HTML5 Video
-# - Launch xterm, kill firefox, cleanup previous firefox configuration, launch
-# firefox
-# - Open "youtube.com/html5" and check result
-# - Open "youtube.com/watch?v=Z4j5rJQMdOU" and check result
-# - Exit firefox
-# Maintainer: wnereiz <wnereiz@gmail.com>
+# Summary: Case#1479221: Firefox: HTML5
+# - Launch xterm, install python and selenium
+# - Open opensuse html5 test page
+# - Access various page elements and check results
+# - Close
+# Maintainer: vanastasiadis <vasilios.anastasiadis@suse.com>
 
 use strict;
 use warnings;
 use base "x11test";
 use testapi;
+use utils;
 
 sub run {
     my ($self) = @_;
-    $self->start_firefox_with_profile;
+    select_console "x11";
+    x11_start_program('xterm');
 
-    $self->firefox_open_url('youtube.com/watch?v=Z4j5rJQMdOU');
-    while (check_screen([qw(firefox-youtube-signin firefox-accept-youtube-cookies)], 15)) {
-        if (match_has_tag('firefox-accept-youtube-cookies')) {
-            # get to the accept button with tab and space
-            wait_still_screen(2);
-            send_key_until_needlematch('firefox-accept-youtube-cookies-agree', 'tab', 7, 1);
-            assert_and_click('firefox-accept-youtube-cookies-agree');
-            wait_still_screen(2);
-            next;
-        }
-        elsif (match_has_tag('firefox-youtube-signin')) {
-            assert_and_click('firefox-youtube-signin');
-            wait_still_screen(2);
-            next;
+    # prepare python + selenium
+    become_root;
+    quit_packagekit;
+    zypper_call("in python3");
+    enter_cmd "exit";
+    assert_script_run("pip3 install selenium");
+    assert_script_run "mkdir temp_selenium && cd temp_selenium";
+    assert_script_run "wget https://github.com/mozilla/geckodriver/releases/download/v0.30.0/geckodriver-v0.30.0-linux64.tar.gz && tar -xf geckodriver-v0.30.0-linux64.tar.gz";
+    assert_script_run "export PATH=\"\$(pwd):\$PATH\"";
+    assert_script_run "wget --quiet " . data_url('selenium/selenium_html5.py') . " -O selenium_html5.py";
+
+    # run selenium script, parse results and upload logs
+    assert_script_run "python3 selenium_html5.py >&1 | tee selenium_output.txt ", timeout => 120;
+    my $res = script_output("cat selenium_output.txt");
+    my @errs = ();
+    foreach my $line (split(/\n/, $res)) {
+        if (index($line, 'FAIL') != -1) {
+            my ($match) = $line =~ /\[\[ ( (?:(?!\]\]).)* ) \]\]/x;
+            push(@errs, $match);
```

Comment:
Maybe we can look into pytest? another idea is to rather evaluate the exit code of the script, anything that is a failure, is more than `0`.

Reply:
Ultimately, I went with neither options, but rather checking the output file in the end. I *could* just let the python script fail and fail the test, but then the output file would not exist and be uploaded for the user to check details at their leisure. Pytest felt wrong for this purpose, see the related ticket for more details on that.